File: practice/src/main/java/com/decade/practice/utils/TokenUtils.java
package com.decade.practice.utils;

import jakarta.servlet.http.HttpServletRequest;

import java.util.Enumeration;

public class TokenUtils {
    public static final String REFRESH_PARAM = "refresh_token";
    public static final String HEADER_NAME = "Authorization";
    public static final String BEARER = "Bearer ";

    public static String extractToken(HttpServletRequest request) {
        Enumeration<String> headers = request.getHeaders(HEADER_NAME);
        while (headers.hasMoreElements()) {
            String header = headers.nextElement();
            String token = extractToken(header);
            if (token != null) {
                return token;
            }
        }
        return null;
    }

    public static String extractToken(String headerValue) {
        if (headerValue != null && headerValue.startsWith(BEARER)) {
            return headerValue.substring(BEARER.length());
        }
        return null;
    }

    public static boolean hasToken(HttpServletRequest request) {
        return extractToken(request) != null || extractRefreshToken(request) != null;
    }

    public static String extractRefreshToken(HttpServletRequest request) {
        return request.getParameter(REFRESH_PARAM);
        // Commented out in original:
        // ?: request.cookies?.find { cookie -> cookie.name == REFRESH_PARAM }?.value
    }
}

File: practice/src/main/java/com/decade/practice/utils/TokenUtils.kt
package com.decade.practice.utils

import jakarta.servlet.http.HttpServletRequest

object TokenUtils {
      const val REFRESH_PARAM: String = "refresh_token"
      const val HEADER_NAME: String = "Authorization"
      const val BEARER: String = "Bearer "

      fun extractToken(request: HttpServletRequest): String? {
            val headers = request.getHeaders(HEADER_NAME)
            for (header in headers) {
                  if (header.startsWith(BEARER)) {
                        return header.substring(BEARER.length)
                  }
            }

            return null
      }

      fun hasToken(request: HttpServletRequest): Boolean =
            extractToken(request) != null || extractRefreshToken(request) != null

      fun extractRefreshToken(request: HttpServletRequest): String? = request.getParameter(REFRESH_PARAM)
//            ?: request.cookies?.find { cookie -> cookie.name == REFRESH_PARAM }?.value
}